tests/Issues/InfiniteLoop/Rector/MethodCall/InfinityLoopRector.php

```diff
@@ -20,15 +21,18 @@ final class InfinityLoopRector extends AbstractRector
      */
     public function getNodeTypes(): array
     {
-        return [MethodCall::class];
+        return [Assign::class, MethodCall::class];
     }
 
     /**
-     * @param MethodCall $node
-     * @return Assign|null
+     * @param Assign|MethodCall $node
      */
-    public function refactor(Node $node): ?Node
+    public function refactor(Node $node): Assign|null|int
     {
+        if ($node instanceof Assign) {
+            return NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN;
+        }
```

Comment:
@TomasVotruba here to avoid road to left to left to left example test :)

Reply:
Looks good :+1: